### ISSUE TITLE
JavaDoc for string prefence fields with empty default value seems incomplete

### ISSUE THREAD

**Issue by ironjan** [androidannotations/androidannotations#1877]
## Issue

The javadoc for `@SharedPref`s is generated in [SharedPrefHolder](https://github.com/androidannotations/androidannotations/blob/b1e1903307d143b22f985d61c8b8e552d9c1c291/AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/holder/SharedPrefHolder.java#L121) like so:

```
if (defaultValueStr != null) {
    fieldMethod.javadoc().append("<p><b>Defaults to</b>: " + defaultValueStr + "</p>\n");
}
```

The following example shows what javadocs are generated. The empty default strings lead to a javadoc that seems incomplete when viewed with Android Studio.

```
@SharedPref
public interface PreferencesInterface {
    // also works with @DefaultString("")
    // will generate "<p><b>Defaults to</b>: </p>"
   String stringField();

   // will generate "<p><b>Defaults to</b>: not empty</p>"
   @DefaultString("not empty")
   String anotherStringField();
}
```

![bildschirmfoto zu 2016-10-16_13-41-52](https://cloud.githubusercontent.com/assets/1698836/19417175/661ebf76-93a6-11e6-8813-c46a32aab793.png)

## Suggestions

One possibility would be to include back-ticks in the generated javadoc. This seems to me like a clean solution; I have seen this in ruby docs but not yet in the "java world". 

```
<p><b>Defaults to</b>: `" + defaultValueStr + "`</p>\n
```

This would generate javadocs like the following:
```
for empty strings:
  <p><b>Defaults to</b>: ``</p>
for @DefaultString("non-empty")
  <p><b>Defaults to</b>: `non-empty`</p>
for an int field as non-String example
  <p><b>Defaults to</b>: `0`</p>
```

Another possibility would be to handle strings so that docs like `<p><b>Defaults to</b>: ""</p>` or `<p><b>Defaults to</b>: "not empty"</p>` are generated.

**Comment by ironjan** [comment]
Pull Request submitted as #1878. If the back-tick solution is not as clean as I thought, feel free to close the PR and let me know the preferred solution.

**Comment by dodgex** [comment]
Fixed in #1878
